File: mobile/lib/stores/toast_store.dart
// lib/stores/toast_store.dart
import 'dart:async';
import 'package:flutter_riverpod/flutter_riverpod.dart';

enum ToastType { success, error, warning, info }

class ToastData {
  final int id;
  final String title;
  final String? description;
  final ToastType type;
  final int duration;

  const ToastData({
    required this.id,
    required this.title,
    this.description,
    this.type = ToastType.info,
    this.duration = 3500,
  });
}

class ToastNotifier extends StateNotifier<List<ToastData>> {
  int _nextId = 0;

  ToastNotifier() : super([]);

  void show({
    required String title,
    String? description,
    ToastType type = ToastType.info,
    int duration = 3500,
  }) {
    final id = ++_nextId;
    final toast = ToastData(
      id: id,
      title: title,
      description: description,
      type: type,
      duration: duration,
    );
    // Keep max 3 visible
    state = [
      ...state.length > 2 ? state.sublist(state.length - 2) : state,
      toast,
    ];
    if (duration > 0) {
      Timer(Duration(milliseconds: duration), () => remove(id));
    }
  }

  void remove(int id) {
    state = state.where((t) => t.id != id).toList();
  }

  void success(String title, {String? description}) =>
      show(title: title, description: description, type: ToastType.success, duration: 2500);

  void error(String title, {String? description}) =>
      show(title: title, description: description, type: ToastType.error, duration: 4000);

  void warning(String title, {String? description}) =>
      show(title: title, description: description, type: ToastType.warning, duration: 3500);

  void info(String title, {String? description}) =>
      show(title: title, description: description, type: ToastType.info, duration: 3000);
}

final toastProvider = StateNotifierProvider<ToastNotifier, List<ToastData>>(
  (ref) => ToastNotifier(),
);

File: web/src/stores/toastStore.js
// frontend/src/stores/toastStore.js
import { create } from "zustand";

let toastId = 0;

export const useToastStore = create((set) => ({
    toasts: [],

    addToast: ({ title, description, type = "info", duration = 3500 }) => {
        const id = ++toastId;

        set((s) => ({
            toasts: [
                ...s.toasts.slice(-4), // Keep max 5 toasts
                { id, title, description, type, duration },
            ],
        }));

        if (duration > 0) {
            setTimeout(() => {
                set((s) => ({ toasts: s.toasts.filter((t) => t.id !== id) }));
            }, duration);
        }

        return id;
    },

    removeToast: (id) =>
        set((s) => ({ toasts: s.toasts.filter((t) => t.id !== id) })),
}));

export function toast(opts) {
    if (typeof opts === "string")
        return useToastStore.getState().addToast({ title: opts });
    return useToastStore.getState().addToast(opts);
}

toast.success = (title, opts) =>
    toast({ title, type: "success", duration: 2500, ...opts });
toast.error = (title, opts) =>
    toast({ title, type: "error", duration: 4000, ...opts });
toast.warning = (title, opts) =>
    toast({ title, type: "warning", duration: 3500, ...opts });
toast.info = (title, opts) =>
    toast({ title, type: "info", duration: 3000, ...opts });